docs/source/general/data.rst

```diff
@@ -2,11 +2,429 @@
 Data
 ####
 
+.. _data:
+
+
+***********
+Terminology
+***********
+
+Here are common terms you need to be familiar with:
+
+.. list-table:: Terminology
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Term
+     - Definition
+   * - :class:`~flash.data.data_module.DataModule`
+     - The :class:`~flash.data.data_module.DataModule` contains the dataset, transforms and dataloaders.
+   * - :class:`~flash.data.data_pipeline.DataPipeline`
+     - The :class:`~flash.data.data_pipeline.DataPipeline` is Flash internal object to manage :class:`~flash.data.process.Preprocess` and :class:`~flash.data.process.Postprocess` objects.
+   * - :class:`~flash.data.process.Preprocess`
+     - The :class:`~flash.data.process.Preprocess` provides a simple hook-based API to encapsulate your pre-processing logic.
+        The :class:`~flash.data.process.Preprocess` provides multiple hooks such as :meth:`~flash.data.process.Preprocess.load_data`
+        and :meth:`~flash.data.process.Preprocess.load_sample` which are used to replace a traditional `Dataset` logic.
+        Flash DataPipeline contains a system to call the right hooks when needed.
+        The :class:`~flash.data.process.Preprocess` hooks covers from data-loading to model forwarding.
+   * - :class:`~flash.data.process.Postprocess`
+     - The :class:`~flash.data.process.Postprocess` provides a simple hook-based API to encapsulate your post-processing logic.
+        The :class:`~flash.data.process.Postprocess` hooks covers from model outputs to predictions export.
+
+*******************************************
+How to use out-of-the-box flashdatamodules
```

Comment:
Move this to the top of the page.

It should be 

**Using built-in DataModules**
flash has built in datamodules you can use to load your data to any existing flash task. For more info, check out....

**Custom Data**
If you are creating a new task or using a different type of dataset for an existing task, you may want to override .... 

and then add the terminology table and everything after that